crates/viewer/re_viewer_context/src/gpu_bridge/image_to_gpu.rs

```diff
@@ -248,19 +271,31 @@ pub fn texture_creation_desc_from_color_image<'a>(
     } else {
         let color_model = image.format.color_model();
         let datatype = image.format.datatype();
+
         let (data, format) = match (color_model, datatype) {
-            // Normalize sRGB(A) textures to 0-1 range, and let the GPU premultiply alpha.
-            // Why? Because premul must happen _before_ sRGB decode, so we can't
+            // sRGB(A) handling is done by `ColormappedTexture`.
+            // Why not use `Rgba8UnormSrgb`? Because premul must happen _before_ sRGB decode, so we can't
             // use a "Srgb-aware" texture like `Rgba8UnormSrgb` for RGBA.
             (ColorModel::RGB, ChannelDatatype::U8) => (
                 pad_rgb_to_rgba(&image.buffer, u8::MAX).into(),
                 TextureFormat::Rgba8Unorm,
             ),
 
+            (ColorModel::BGR, ChannelDatatype::U8) => {
+                (
+                    pad_rgb_to_rgba(&image.buffer, u8::MAX).into(),
+                    TextureFormat::Bgra8Unorm,
```

Comment:
Is using the `Bgra` texture even a good idea, or is it just adding complexity?

Reply:
from the pov of the all knowing rectangle shader it's just extra complexity, so I should add a comment for the rationale of doing this anyways!

The reason is that this allow us to use BGR(A) on meshes and other occasions where we today don't have fancy decoding etc..
This is very very subtly hinted at by the behavior of `required_shader_decode` which the mesh visualizer uses to determine if it should warn the user that it can't consume a given image.
(in the glorious 🌈 future 🌈  re_renderer will have an internal conversion pipeline that will allow us to do texture conversions separately and gpu accelerated on the fly for all types. But we live in the now and gotta squeeze out the 80% with what we have 😄)

Reply:
for the code, cleaned up and more formalized :)
```
            // Make use of wgpu's BGR(A)8 formats.
            //
            // From the pov of our on-the-fly decoding textured rect shader this is just a strange special case
            // given that it already has to deal with other BGR(A) formats.
            //
            // However, we have other places where we don't have the luxury of having a shader that can do the decoding for us.
            // In those cases we'd like to support as many formats as possible without decoding.
            //
            // (in some hopefully not too far future, re_renderer will have an internal conversion pipeline
            // that injects on-the-fly texture conversion from source formats before the consumer of a given texture is run
            // and caches the result alongside with the source data)
            //
            // See also [`required_shader_decode`] which lists this case as a format that does not need to be decoded.
```